cli/command/cli.go

```diff
@@ -175,9 +175,26 @@ func (cli *DockerCli) RegistryClient(allowInsecure bool) registryclient.Registry
 	return registryclient.NewRegistryClient(resolver, UserAgent(), allowInsecure)
 }
 
+// WithInitializeOpt is the type of the functional options passed to DockerCli.Initialize
+type WithInitializeOpt func(dockerCli *DockerCli) error
+
+// WithInitilizeClient is passed to DockerCli.Initialize by callers who wish to set a particular API Client for use by the CLI.
+func WithInitilizeClient(makeClient func(dockerCli *DockerCli) (client.APIClient, error)) WithInitializeOpt {
```

Comment:
typo 😅 

```suggestion
func WithInitializeClient(makeClient func(dockerCli *DockerCli) (client.APIClient, error)) WithInitializeOpt {
```

Reply:
Oops! The perils of enabling autocomplete means these things get propagated!